File: 백준/Silver/2485. 가로수/가로수.py
import math

n = int(input())
arr = []
distance = []

for i in range(n):
    num = int(input())
    arr.append(num)
for i in range(n - 1):
    num = arr[i + 1] - arr[i]
    distance.append(num)

min = distance[0]
for i in range(1, n - 1):
    min = math.gcd(min, distance[i])

total = 0
for i in range(n - 1):
    total += distance[i] // min - 1
print(total)

File: 백준/Silver/2485. 가로수/가로수.c
#include <stdio.h>
#include <stdlib.h>

int gcd(int a, int b);

int main(void) {
	int n, min = 1000000001, cnt = 0, GCD;
	scanf("%d", &n);
	int* arr = (int*)malloc(sizeof(int) * n);
	int* num = (int*)malloc(sizeof(int) * (n - 1));
	for (int i = 0; i < n; i++) {
		scanf("%d", &arr[i]);
	}

	for (int i = 0; i < n - 1; i++) {
		num[i] = arr[i + 1] - arr[i];
	}
	GCD = num[0];
	for (int i = 1; i < n - 1; i++){
		GCD = gcd(GCD, num[i]);
	}
	for (int i = 0; i < n - 1; i++) {
		cnt += (num[i] / GCD - 1);
	}
	printf("%d", cnt);


	return 0;
}

int gcd(int a, int b) {
	if (b == 0) return a;
	else return gcd(b, a % b);
}